### DISCUSSION TITLE
Data Importer taking up 100% of CPUs

### DISCUSSION THREAD

**Discussion by atjbramley** [firefly-iii/firefly-iii#6738] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Data importer is causing permanent 100% CPU usage on the machine I have running it.  Previously it took absolutely minimal resources.  

The apache2-DFOREGROUND process is causing the issue - I can confirm this with HTOP.
Stopping the data-importer docker container also causes CPU usage to return to normal.

A sample of logs are attached here (similar to extra debug info)
https://gist.github.com/atjbramley/3b8d079e612a580a1053afd90c258a56

I don't recall making any changes to my ff_iii setup that would have caused this.

The data importer seems to work despite this!  

### Expected behaviour

Data importer should hardly use any CPU - especially when not working.

### Debug information

Debug information generated at 2022-12-22 17:44:57 Europe/Amsterdam for Firefly III Data Importer version **0.9.17**.

| Scope | Version |
| --- | --- |
| FIDI | 0.9.17 |
| Build | 177, 23-10-2022 15:04:13 CEST |
| PHP | 8.1.11 |
| Host | Linux |

| System info | Value |
| --- | --- |
| Using docker? | true |
| System TZ |  |
| Browser TZ | Europe/Rome |
| App environment | production |
| App debug mode | false |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Interface | apache2handler |
| BCscale | 12 |
| Trusted proxies (.env) |  |

| User info | Value |
| --- | --- |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:108.0) Gecko/20100101 Firefox/108.0 |

### Steps to reproduce

1. Run data importer docker container

## Steps attempted

- Tried re-authenticating by generating new access token
- Tried pulling new image and restarting containers
- Tried adding 'depends-on firefly_iii' to docker-compose (data importer depends on ff_iii)
- Tried deleting container and recreating from image

**Comment by JC5** [comment]
Check the search. In some cases your container manager may hamer the web server. I've seen it before.

**Comment by JC5** [comment]
Yep. In your logs, check out these lines:

```
172.21.0.1 - - [22/Dec/2022:16:39:11 +0000] "GET / HTTP/1.1" 302 1517 "-" "axios/0.27.2"
```

Somebody (using axios) keeps hammering the index.

**Comment by atjbramley** [comment]
Awesome - I thought that was outbound, not inbound.  You were right - I had another service going nuts for the data importer.

Maybe it is interesting though, because the same service was scanning everything running in docker (15 containers or so), and only data importer behaved like that?

Thanks for your help 👍

**Reply by JC5** [reply]
The index tries to grab a token from Firefly III and does a short oAuth2 cycle. That can be resource intensive for both containers.

**Reply by atjbramley** [reply]
Understood!  Well, at least it was keeping the house a little warmer in this chilly winter!
